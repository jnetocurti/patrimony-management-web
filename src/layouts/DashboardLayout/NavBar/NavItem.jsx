import React, { useState, Fragment } from 'react';
import { NavLink as RouterLink } from 'react-router-dom';

import {
  Button,
  Collapse,
  List,
  ListItem,
  makeStyles
} from '@material-ui/core';
import { ExpandLess, ExpandMore } from '@material-ui/icons';

import clsx from 'clsx';
import PropTypes from 'prop-types';

const useStyles = makeStyles((theme) => ({
  item: {
    display: 'flex',
    paddingTop: theme.spacing(0),
    paddingBottom: theme.spacing(0)
  },
  nested: {
    display: 'flex',
    paddingTop: theme.spacing(0),
    paddingBottom: theme.spacing(0),
    paddingLeft: theme.spacing(2)
  },
  button: {
    color: theme.palette.text.secondary,
    fontWeight: theme.typography.fontWeightMedium,
    justifyContent: 'flex-start',
    letterSpacing: theme.spacing(0),
    padding: '8px 6px',
    textTransform: 'none',
    width: '100%'
  },
  icon: {
    marginRight: theme.spacing(1)
  },
  title: {
    marginRight: 'auto'
  },
  active: {
    color: theme.palette.primary.main,
    '& $title': {
      fontWeight: theme.typography.fontWeightMedium
    },
    '& $icon': {
      color: theme.palette.primary.main
    }
  }
}));

const NavItem = ({ config }) => {
  const classes = useStyles();
  const [open, setOpen] = useState(config.open || false);

  const handleClick = () => setOpen(!open);

  const createItem = (options) => {
    const Icon = options.icon;
    const itemClass = options.isParent ? classes.item : classes.nested;
    return (
      <>
        <ListItem
          disableGutters
          className={clsx(itemClass)}
          onClick={options.handleClick}
        >
          <Button
            to={options.href}
            component={RouterLink}
            className={classes.button}
            activeClassName={classes.active}
          >
            {Icon && <Icon className={classes.icon} size="20" />}
            <span className={classes.title}>{options.title}</span>
          </Button>
          {options.isParent
            && options.subItems
            && (open ? <ExpandLess /> : <ExpandMore />)}
        </ListItem>
        {options.subItems && (
          <Collapse in={open} timeout="auto" unmountOnExit>
            <List component="div" disablePadding>
              {options.subItems.map((item) => (
                <Fragment key={item.title}>
                  {createItem({
                    isParent: false,
                    href: item.href,
                    icon: item.icon,
                    title: item.title
                  })}
                </Fragment>
              ))}
            </List>
          </Collapse>
        )}
      </>
    );
  };

  return (
    <>
      {createItem({
        handleClick,
        href: config.href,
        icon: config.icon,
        title: config.title,
        subItems: config.subItems,
        isParent: !!config.subItems
      })}
    </>
  );
};

NavItem.propTypes = {
  config: PropTypes.shape({
    open: PropTypes.bool,
    href: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    icon: PropTypes.elementType.isRequired,
    subItems: PropTypes.arrayOf(
      PropTypes.shape({
        href: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        icon: PropTypes.elementType.isRequired
      })
    )
  })
};

export default NavItem;
