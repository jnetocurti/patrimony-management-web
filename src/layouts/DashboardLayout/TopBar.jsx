import React, { useState } from 'react';
import { Link as RouterLink } from 'react-router-dom';

import {
  AppBar,
  Badge,
  Box,
  Hidden,
  IconButton,
  Toolbar,
  makeStyles
} from '@material-ui/core';
import InputIcon from '@material-ui/icons/Input';
import MenuIcon from '@material-ui/icons/Menu';
import NotificationsIcon from '@material-ui/icons/NotificationsOutlined';

import clsx from 'clsx';
import PropTypes from 'prop-types';
import Logo from 'src/components/Logo';

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: theme.palette.background.default
  },
  avatar: {
    width: 60,
    height: 60
  }
}));

const TopBar = ({ className, onMobileNavOpen, ...rest }) => {
  const classes = useStyles();
  const [notifications] = useState([]);

  return (
    <AppBar className={clsx(classes.root, className)} elevation={0} {...rest}>
      <Toolbar>
        <RouterLink to="/">
          <Logo />
        </RouterLink>
        <Box flexGrow={1} />
        <Hidden mdDown>
          <IconButton color="primary">
            <Badge
              badgeContent={notifications.length}
              color="default"
              variant="dot"
            >
              <NotificationsIcon />
            </Badge>
          </IconButton>
          <IconButton color="primary">
            <InputIcon />
          </IconButton>
        </Hidden>
        <Hidden lgUp>
          <IconButton color="default" onClick={onMobileNavOpen}>
            <MenuIcon />
          </IconButton>
        </Hidden>
      </Toolbar>
    </AppBar>
  );
};

TopBar.propTypes = {
  className: PropTypes.string,
  onMobileNavOpen: PropTypes.func
};

export default TopBar;
