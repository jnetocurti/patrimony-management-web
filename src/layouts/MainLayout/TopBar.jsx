import React from 'react';
import { Link as RouterLink } from 'react-router-dom';

import { AppBar, Toolbar, makeStyles } from '@material-ui/core';

import clsx from 'clsx';
import PropTypes from 'prop-types';
import Logo from 'src/components/Logo';

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: theme.palette.background.default
  },
  toolbar: {
    height: 64
  }
}));

const TopBar = ({ className, ...rest }) => {
  const classes = useStyles();

  return (
    <AppBar className={clsx(classes.root, className)} elevation={0} {...rest}>
      <Toolbar className={classes.toolbar}>
        <RouterLink to="/">
          <Logo />
        </RouterLink>
      </Toolbar>
    </AppBar>
  );
};

TopBar.propTypes = {
  className: PropTypes.string
};

export default TopBar;
