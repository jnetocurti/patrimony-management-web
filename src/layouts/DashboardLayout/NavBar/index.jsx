import React, { useEffect } from 'react';
import { useLocation } from 'react-router-dom';

import {
  Box,
  Divider,
  Drawer,
  Hidden,
  List,
  makeStyles
} from '@material-ui/core';
import { Apartment as ApartmentIcon, ListOutlined } from '@material-ui/icons';

import PropTypes from 'prop-types';

import NavItem from './NavItem';

const items = [
  {
    open: true,
    icon: ApartmentIcon,
    title: 'Fundos imobiliários',
    href: '/app/real-estate-funds/ranking',
    subItems: [
      {
        icon: ListOutlined,
        title: 'Ranking',
        href: '/app/real-estate-funds/ranking'
      },
      {
        icon: ListOutlined,
        title: 'Transações',
        href: '/app/real-estate-funds/transactions'
      }
    ]
  }
];

const useStyles = makeStyles(() => ({
  mobileDrawer: {
    width: 225
  },
  desktopDrawer: {
    top: 64,
    width: 225,
    height: 'calc(100% - 64px)'
  }
}));

const NavBar = ({ onMobileClose, openMobile }) => {
  const classes = useStyles();
  const location = useLocation();

  useEffect(() => {
    if (openMobile && onMobileClose) {
      onMobileClose();
    }
  }, [location.pathname]);

  const content = (
    <Box height="100%" display="flex" flexDirection="column">
      <Divider />
      <Box p={2}>
        <List>
          {items.map((item) => (
            <NavItem key={item.title} config={item} />
          ))}
        </List>
      </Box>
      <Box flexGrow={1} />
    </Box>
  );

  return (
    <>
      <Hidden lgUp>
        <Drawer
          anchor="left"
          classes={{ paper: classes.mobileDrawer }}
          onClose={onMobileClose}
          open={openMobile}
          variant="temporary"
        >
          {content}
        </Drawer>
      </Hidden>
      <Hidden mdDown>
        <Drawer
          anchor="left"
          classes={{ paper: classes.desktopDrawer }}
          open
          variant="persistent"
        >
          {content}
        </Drawer>
      </Hidden>
    </>
  );
};

NavBar.propTypes = {
  onMobileClose: PropTypes.func,
  openMobile: PropTypes.bool
};

NavBar.defaultProps = {
  onMobileClose: () => {},
  openMobile: false
};

export default NavBar;
