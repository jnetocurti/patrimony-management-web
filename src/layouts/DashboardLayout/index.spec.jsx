import React from 'react';

import { shallow } from 'enzyme';

import DashboardLayout from '.';
import NavBar from './NavBar';
import TopBar from './TopBar';

describe('src/components/Logo', () => {
  it('should render correctly', () => {
    const component = shallow(<DashboardLayout />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly on mobile nav open', () => {
    const component = shallow(<DashboardLayout />);
    component.find(TopBar).simulate('mobileNavOpen');
    expect(component).toMatchSnapshot();
  });

  it('should render correctly on mobile close', () => {
    const component = shallow(<DashboardLayout />);
    component.find(NavBar).simulate('mobileClose');
    expect(component).toMatchSnapshot();
  });
});
