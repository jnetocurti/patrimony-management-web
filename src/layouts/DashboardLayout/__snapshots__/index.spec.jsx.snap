// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Logo should render correctly 1`] = `
<Store>
  <div
    className="makeStyles-root-1"
  >
    <TopBar
      onMobileNavOpen={[Function]}
    />
    <NavBar
      onMobileClose={[Function]}
      openMobile={false}
    />
    <div
      className="makeStyles-wrapper-2"
    >
      <div
        className="makeStyles-contentContainer-3"
      >
        <div
          className="makeStyles-content-4"
        >
          <Outlet />
        </div>
      </div>
    </div>
  </div>
</Store>
`;

exports[`src/components/Logo should render correctly on mobile close 1`] = `
<Store>
  <div
    className="makeStyles-root-1"
  >
    <TopBar
      onMobileNavOpen={[Function]}
    />
    <NavBar
      onMobileClose={[Function]}
      openMobile={false}
    />
    <div
      className="makeStyles-wrapper-2"
    >
      <div
        className="makeStyles-contentContainer-3"
      >
        <div
          className="makeStyles-content-4"
        >
          <Outlet />
        </div>
      </div>
    </div>
  </div>
</Store>
`;

exports[`src/components/Logo should render correctly on mobile nav open 1`] = `
<Store>
  <div
    className="makeStyles-root-1"
  >
    <TopBar
      onMobileNavOpen={[Function]}
    />
    <NavBar
      onMobileClose={[Function]}
      openMobile={true}
    />
    <div
      className="makeStyles-wrapper-2"
    >
      <div
        className="makeStyles-contentContainer-3"
      >
        <div
          className="makeStyles-content-4"
        >
          <Outlet />
        </div>
      </div>
    </div>
  </div>
</Store>
`;
